src/app/collective-rewards/rewards/components/ClaimRewardModal/ClaimRewardsModalView.tsx

```diff
@@ -0,0 +1,167 @@
+  subLabel: ReactNode
+  isLoading?: boolean
+}
+
+const ClaimRewardRadioOption: FC<ClaimRewardRadioOptionProps> = memo(
+  ({ value, label, subLabel, isLoading }) => (
+    <RadioGroup.Item
+      value={value}
+      className="flex-1 rounded border-1 border-[v3-text-100] px-3 py-4 flex flex-col items-start gap-2 data-[state=checked]:border-t-4 data-[state=checked]:pt-[13px] focus:outline-none cursor-pointer"
+    >
+      {isLoading ? (
+        <div className="flex justify-center items-center w-full py-2">
+          <LoadingSpinner size="small" />
+        </div>
+      ) : (
+        <div className="flex flex-row items-start gap-3 w-full">
+          <span className="w-5 h-5 aspect-square rounded-full border-2 border-white flex items-center justify-center flex-shrink-0">
+            <RadioGroup.Indicator className="w-full h-full rounded-full border-4 border-white" />
+          </span>
+          <div className="flex flex-col items-start gap-2 justify-start text-left w-full">
+            <Typography variant="h3">{label}</Typography>
+            <Typography variant="body" className="text-v3-bg-accent-0">
+              {subLabel}
+            </Typography>
+          </div>
+        </div>
+      )}
+    </RadioGroup.Item>
+  ),
+)
+
+ClaimRewardRadioOption.displayName = 'ClaimRewardRadioOption'
+
+export const ClaimRewardsModalView: FC<ClaimRewardsModalViewProps> = ({
+  onClose,
+  selectedRewardType,
+  onRewardTypeChange,
+  rifAmount,
+  rbtcAmount,
+  rifFiatAmount,
+  rbtcFiatAmount,
+  totalFiatAmount,
+  onClaim,
+  isClaimable,
+  isTxPending = false,
+  isLoading = false,
+  className,
+}) => {
+  const radioOptions: Array<{
+    value: ClaimRewardType
+    label: ReactNode
+    subLabel: ReactNode
+  }> = [
+    {
+      value: 'all',
+      label: 'All Rewards',
+      subLabel: formatCurrency(totalFiatAmount),
+    },
+    {
+      value: 'rif',
+      label: (
+        <div className="flex items-center gap-2">
+          {formatSymbol(rifAmount, RIF)} <TokenImage symbol={TokenSymbol.STRIF} size={16} />
+        </div>
+      ),
+      subLabel: `${formatCurrency(rifFiatAmount)} USD`,
+    },
+    {
+      value: 'rbtc',
+      label: (
+        <div className="flex items-center gap-2">
+          {formatSymbol(rbtcAmount, RBTC)} <TokenImage symbol={TokenSymbol.RBTC} size={18} />
+        </div>
+      ),
+      subLabel: formatCurrency(rbtcFiatAmount),
+    },
+  ]
+
+  return (
+    <Modal
+      onClose={onClose}
+      width={700}
+      height="auto"
+      closeButtonColor="white"
+      data-testid="ClaimRewardsModalView"
+      className={cn('font-rootstock-sans shadow-[0px_0px_40px_0px_rgba(255,255,255,0.10)]', className)}
+    >
+      <div className="p-8 flex flex-col gap-8">
+        <Typography variant="h1">CLAIM REWARDS</Typography>
+        <Typography>
+          Select the rewards that you want to claim, then confirm the transaction in your wallet.
+        </Typography>
+        <RadioGroup.Root
```

Comment:
Im trying the buttons but are not working, when I pressed it, the radio does not change.